### PR TITLE
fix(kit): `Tab` redundant space after the last tab

### LINKED ISSUE #2002: 🐞 - `Tab` redundant `margin-right` after the last tab
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://taiga-ui.dev/navigation/tabs#desktop

### Is this issue blocking you?

Non-Blocking

### Description

1. https://taiga-ui.dev/navigation/tabs#desktop
taigaDevTab

2. Our internal project
ourInternalProject

### Angular version

any

### Taiga UI version

2.50.2

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?
Closes #2002

## What is the new behavior?
01- wide-screen - no-margin-last-tab diff

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/kit/components/tabs/tabs/tabs.style.less [hunk 1]
```diff
@@ -21,4 +21,11 @@
         background: var(--tui-link);
         box-shadow: none;
     }
+
+    &::ng-deep > :where(:not(:last-child)) {
+        & [tuiTab],
+        &[tuiTab] {
```
Comment (nsbarsukov) [inline projects/kit/components/tabs/tabs/tabs.style.less:27]: awful solution :(

Guys, do you have any ideas how to improve it ?

Comment (vladimirpotekhin) [reply projects/kit/components/tabs/tabs/tabs.style.less:27]: maybe just add compensation in tabs-with-more? or problem is appears not only in tabs-with-more ?

Comment (nsbarsukov) [reply projects/kit/components/tabs/tabs/tabs.style.less:27]: Current solution works. But I agree that it is awful(

Comment (splincode) [reply projects/kit/components/tabs/tabs/tabs.style.less:27]: Can we write e2e tests for `example, [tuiTab] in wrapped by hosted-dropdown`? 
and add `TODO: find the better way.` for `&::ng-deep`

**Comment by splincode** [comment]
Why e2e tests didn't fail?

**Comment by splincode** [comment]
Oh, sorry, wrong button

File: projects/kit/components/tabs/tabs/tabs.style.less (hunk 1 above)
Comment (nsbarsukov) [reply projects/kit/components/tabs/tabs/tabs.style.less:27]: @splincode 
> Can we write e2e tests for example, [tuiTab] is wrapped by hosted-dropdown

we have already this test (`demo.spec.ts`)

**Comment by nsbarsukov** [comment]
> Why e2e tests didn't fail?

No tests fails => this PR does not break anything (I hope, at least :smile:)

But we don't has the test for this bug)
I will add test for this case 👌

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/kit/components/tabs/tabs/tabs.style.less (hunk 1 above)
Comment (splincode) [reply projects/kit/components/tabs/tabs/tabs.style.less:27]:
```
But we don't have the test for this bug)
I will add test for this case 👌
```

Thank you))

**Comment by nsbarsukov** [comment]
@splincode @vladimirpotekhin I added cypress test for this bug

**Comment by waterplea** [comment]
Now I wonder if we need `TAB_MARGIN` at all, maybe just keep it a CSS variable — it can easily be overridden in new `Navigation` component and even done so dynamically, depending on the resolution. Also I think we can simplify everything by making it `margin-left` instead of right and this way we can easily eliminate first one by `tui-tab > :fist-child`

File: projects/demo-integrations/cypress/tests/kit/tabs/tabs.spec.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+describe('Tabs', () => {
+    it('No extra margin after the last tab', () => {
+        cy.viewport(1700, 900);
+        cy.tuiVisit('/navigation/tabs');
+
+        cy.get('#complex')
+            .findByAutomationId(EXAMPLE_ID)
+            .tuiScrollIntoView()
+            .matchImageSnapshot('01-[wide-screen]-[no-margin-last-tab]');
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/demo-integrations/cypress/tests/kit/tabs/tabs.spec.ts:11]: nope.
The test is excelent. Bad code)

File: projects/kit/components/tabs/tabs-with-more/tabs-with-more.style.less [hunk 3]
```diff
@@ -36,6 +36,7 @@
 
 .t-more_wrapper {
     height: 100%;
+    margin-left: 1.5rem;
```
Comment (nsbarsukov) [inline projects/kit/components/tabs/tabs-with-more/tabs-with-more.style.less:39]: move it before class `t-overflown`

File: projects/kit/components/tabs/tab/tab.style.less [hunk 4]
```diff
@@ -4,6 +4,13 @@
     box-shadow: inset 0 -2px var(--tui-base-03);
 }
 
+:host-context(tui-tabs >):first-child,
+:host-context([tuiTabs] >):first-child,
+:host-context(tui-tabs > :first-child),
+:host-context([tuiTabs] > :first-child) {
+    margin-left: 0;
```
Comment (splincode) [inline projects/kit/components/tabs/tab/tab.style.less:11]: You are great warrior

**Comment by nsbarsukov** [comment]
@waterplea @splincode review again, please

**Review by waterplea — APPROVED** [review]

**Pull request merged**